interfaces/devdoc/malloc_multi_flex_requirements.md

```diff
@@ -0,0 +1,124 @@
+# `MALLOC_MULTI_FLEX` requirements
+
+## Overview
+
+`MALLOC_MULTI_FLEX` allows to allocate memory for a structure containing multiple variable sized array members using a single malloc.
+
+## Exposed API
+
+```c
+#define DEFINE_MALLOC_MULTI_FLEX(type, ...)
+    ...
+
+#define MALLOC_MULTI_FLEX(type)
+    ...
+```
+
+## Example - 
+
+Consider there are two structs as defined below -
+
+```c
+typedef struct INNER_STURCT_TAG
+{
+    uint32_t inner_int_1;
+    uint64_t inner_int_2;
+}INNER_STRUCT;
+
+typedef struct PARENT_STRUCT_TAG
+{
+    uint64_t int_1;
+    uint32_t* array_1;
+    uint32_t int_2;
+    uint64_t* array_2;
+    INNER_STRUCT* array_3;
+    uint32_t int_3;
+}PARENT_STRUCT;
+```
+
+Lets say the user wants to allocate memory for `PARENT_STRUCT` which has three array members:
+- array_1 of type `uint32_t`
+- array_2 of type `uint64_t`
+- array_3 of type `INNER_STRUCT`
+
+The caller first needs to specify the struct and member details using `DEFINE_MALLOC_MULTI_FLEX` macro in `.c` file to create a custom `malloc_multi_flex_<type>` function as defined below -
+
+```c
+DEFINE_MALLOC_MULTI_FLEX(PARENT_STRUCT, array_1, array_2, array_3)
```

Comment:
if we actually include the type of each array field in this macro, then it can internally generate the `sizeof(type)` and `alignof(type)` code and the generated function could just take the array counts.

/consider